timemachine/cpp/src/barostat.cu

```diff
@@ -57,7 +58,7 @@ MonteCarloBarostat::MonteCarloBarostat(
     }
 
     curandErrchk(curandCreateGenerator(&cr_rng_, CURAND_RNG_PSEUDO_DEFAULT));
-    cudaSafeMalloc(&d_rand_, 2 * sizeof(*d_rand_));
+    cudaSafeMalloc(&d_rand_, RANDOM_BATCH_SIZE * 2 * sizeof(*d_rand_));
```

Comment:
nit: do benchmarks show significant improvement from generating random numbers in batches? It does add a little bit of complexity.

Reply:
The wall clock times do not show a difference, but the amount of time running these kernels does reduce.

This went from 180us to 2us in terms of the total kernel time. The time it takes the kernel to generate 2000 values is the same as generating 2 values. Not noticeable with how slow the potentials are.

Could remove this as it doesn't make a difference to wall clock time, just trying to reduce overheads where possible.